neo/core/block.py

```diff
@@ -86,9 +89,27 @@ def __init__(self, name=None, description=None, file_origin=None,
         self.file_datetime = file_datetime
         self.rec_datetime = rec_datetime
         self.index = index
-        self.regionsofinterest = []   # temporary workaround.
-        # the goal is to store all sub-classes of RegionOfInterest in a single list
-        # but this will need substantial changes to container handling
+        self._segments = ObjectList(Segment, parent=self)
+        self._groups = ObjectList(Group, parent=self)
+        self._regionsofinterest = ObjectList(RegionOfInterest, parent=self)
```

Comment:
Should regions of interest really be linked to a block and not a handled on the same level as a view?